# EEP/IAS 118 Assignment 4
Due on November 14th, 2022 at 11.59pm on Gradescope


Trees are able to absorb a lot of the CO2 in the atmosphere. Thus, a lot of policy attention has been focused on the role trees can play in slowing down the increase in CO2 levels. One potential avenue has been to reduce deforestation, by preventing forest owners from cutting down trees. In the context we study and many other contexts, tree cutting is a very lucrative activity (mostly from selling timber), but also provides firewood for fuel and cooking. People might resort to tree cutting as their main economic activity, or to provide them with extra income in case of an emergency. 

One paper by Jayachandran, de Laat, Lambin, Stanton, Audy and Thomas (2017) looks at a policy in Uganda, where farmers in different villages are enrolled into a randomized controlled trial and are offered cash  in exchange of not cutting down their trees (this type of intervention or policy is often called payment for ecosystem services or PES). However there is still debate over whether these are effective (especially cost-effective) policies for environmental conservation. (The paper is [Cash for carbon: A randomized trial of payments for ecosystem services to reduce deforestation](https://www-science-org.libproxy.berkeley.edu/doi/full/10.1126/science.aan0568#supplementary-materials) _Science_ Vol 357, Issue 6348.)

We are going to work with a (simplified) version of their data for this problem set to estimate the effects of cash incentives on deforestation. Compared to the published paper, we also look at a simplified version of the treatment effects.

### Part A:

**Suppose the researchers did not conduct a randomized controlled trial. Instead they surveyed farmers and compared tree cover in villages where village leaders decided to give financial incentives to farmers to stop cutting down trees, compared to villages where no such program was in place. Would this comparison allow you to estimate the effect of cash incentives on environmental conservation? Explain why or why not using both words and potential outcomes notation. 
How would the expression for the estimated impact of cash incentives on tree cover differ if cash incentives were randomized?**

a) No, because this will not give the real interpretation of the ideal world data model since MLR4 is being violated due to no randomization. So no you can't estiamte the effect of cash incentive on enviromental conservation as there is selection Bias that exist due to not being randomzied.

b) If the researchers radomized correctly, the difference between controlled and treatment group wouldn't be statistically different (expectation), if there is any difference its because of treatment rather than the difference between treatment and control group.

c) If randomization is properly done, the two groups should not
be statistically different (in expectation) → any difference can
be attributed solely to the intervention/treatment, rather than
differences between the treatment and control groups


### Part B:

Now let's analyze the RCT data. The dataset contains observations of sixteen variables from 995 farm households. The variables are as follows:

- $partid$: This is just the unique ID for each farmer
- $forest\_cover$: This is the forest cover tree measured in hectares. The tree cover helps us measure the extent of the deforestation. A higher tree cover corresponds to more trees/higher tree density.
- $cash$: This is the treatment variable of interest - equal to 1 if the household was given cash to not cut trees (i.e. in the treatment group) and 0 otherwise (i.e in the control group)
- $cut\_cult$: This is a dummy equal to 1 if households cut trees to clear area for agriculture and 0 otherwise
- $cut\_timber$: This is a dummy equal to 1 if households cut trees for timber products and 0 otherwise 
- $cut\_emerg$: This is a dummy equal to 1 if households cut trees for emergency/large expenses and 0 otherwise
- $any\_tree\_cut\_1yr$: This is a dummy equal to 1 if households cut any tree during the program and 0 otherwise
- $any\_tree\_cut\_3yr$: This is a dummy equal to 1 if households cut any tree in the past 3 years and 0 otherwise
- $timvisit\_y$: This is a dummy equal to 1 if the household received any visit from timber dealers looking to buy trees in the past year and 0 otherwise 
- $opi\_hurtenv\_agr$: This is a dummy equal to 1 if the household head agrees that he has to damage the environment to improve his life and 0 otherwise.
- $envprog\_bi$: This is a dummy equal to 1 if the household head has been involved in any environmental organization and 0 otherwise.
- $log\_tim\_price$: This is the log of total income from cutting trees earned by the household.
- $hhead\_age$: This is the age of the household head
- $hhead\_educ$: This is the number of years of education of the household head
- $hhland\_area$: This is the land area owned by the household, in hectares (ha)
- $hhforest\_area$: This is the forest area owned by the household, in hectares (ha)

Note that all variables other than tree cover and cutting trees during the program were measured *before* the distribution of incentives. They were measured either the year before or for the 3 years before the intervention.

a) **Before running any regressions, show how you can obtain the average treatment effect (ATE) of cash incentives on tree cover. Then write down the regression you could use to estimate the ATE.**

To obtain the average effect (ATE) of cash incentives on tree cover, you would use this regression: 

Forest_cover = B0 + B1T + U

E[Forest_cover^T|cash = 1] - E[Forest_cover^C|cash = 0] --> (True effect)

forest_cover ~ Cash == 1 (Treatment group) - cash == 1 (Control group) --> (code)

**b) Read in the data from `ps4_pes_data.dta` (remember you need to use the `read_dta` function from the `haven` package to read `.dta` files. You will also probably want to load `tidyverse`).** 

**Then, estimate the regression you wrote down using `lm()` and test whether the ATE is statistically significant at a 99\% confidence level.** 

**Is the ATE on cutting trees also statistically significant at the 99\% level? Show your work and interpret the coefficient**

In [4]:
# code here
library(haven) 
library(tidyverse)
data <- read_dta('ps4_pes_data.dta')
reg1 <- lm(formula = forest_cover ~ cash, data = data)
summary(reg1)


Call:
lm(formula = forest_cover ~ cash, data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-12.0697  -0.0676   0.1363   0.3175  17.2493 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) -0.31889    0.05840  -5.460 6.01e-08 ***
cash         0.18257    0.08174   2.234   0.0257 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.289 on 993 degrees of freedom
Multiple R-squared:  0.004999,	Adjusted R-squared:  0.003997 
F-statistic: 4.989 on 1 and 993 DF,  p-value: 0.02573


In [5]:
#code here for second ATE
reg2 <- lm(formula = any_tree_cut_1y ~ cash, data = data)
summary(reg2)


Call:
lm(formula = any_tree_cut_1y ~ cash, data = data)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.4634 -0.4634 -0.3030  0.5366  0.6970 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.46336    0.02229  20.791  < 2e-16 ***
cash        -0.16033    0.03155  -5.081 4.53e-07 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.4801 on 924 degrees of freedom
  (69 observations deleted due to missingness)
Multiple R-squared:  0.02719,	Adjusted R-squared:  0.02613 
F-statistic: 25.82 on 1 and 924 DF,  p-value: 4.532e-07


1. If |t-value| > |critical value|,  We reject the null hypthoesis. Forest cover's T value is 2.234 which is smaller than critical value of 2.581 at 99% confidence level, so we fail to reject the null hypthoesis, meaning there is no statistical evidence for 1% level. 

2. If |t-value| > |critical value|,  We reject the null hypthoesis. Cutting Tree's T value is |-5.081|, which is greather than the critical value of 2.581 at 99% confidence level. We reject the null hypothesis, that is statistical evidence for 1% level.

### Part C:
**a) Now let's check to make sure our sample is balanced across treatment and control. For each of the four household demographics variables, conduct a t-test against the null that they are on average equal between treatment and control.** 

**Coding Hints**: The command for a t-test is `t.test()`. One way to test whether coviarate $X$ is correlated with treatment status $T$, you can put `X~T` inside `t.test()`, just like you would for `lm()`. If you just want to print the t-stat or the p-value rather than the entire test output, you can call `$statistic` or `$p.value`, respectively. 

You can repeat this separately for each covariate, but if you want to be extra fancy you can use the `lapply()` function. `lapply()` is super useful for applying a function repeatedly over different variables. The way you would use it here is `lapply(X, function (x) FUN)` where `X` is the data frame of the variables you are interested in testing and `FUN` is the function you want to apply. One of the arguments of this function should be (small) `x`, which serves as a stand-in for the column of (big) `X` that you want to apply the function over. Finally, if you want to show the output as a single data frame (instead of a list), you can wrap all of this with `as.data.frame()`. With all of this information, you could theoretically produce a nice-ish table of p-values for the four tests you run using a single line of code. You are not required or expected to use this method, but it might come in handy if you find yourself having to work with larger datasets in the future.

In [6]:
#code here
t.test(data[data$cash==0,]$hhead_age,data[data$cash==1,]$hhead_age)
t.test(data[data$cash==0,]$hhead_educ,data[data$cash==1,]$hhead_educ)
t.test(data[data$cash==0,]$hhland_area,data[data$cash==1,]$hhland_area)
t.test(data[data$cash==0,]$hhforest_area,data[data$cash==1,]$hhforest_area)


	Welch Two Sample t-test

data:  data[data$cash == 0, ]$hhead_age and data[data$cash == 1, ]$hhead_age
t = -0.68483, df = 980.28, p-value = 0.4936
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -2.386753  1.151854
sample estimates:
mean of x mean of y 
 47.23265  47.85010 



	Welch Two Sample t-test

data:  data[data$cash == 0, ]$hhead_educ and data[data$cash == 1, ]$hhead_educ
t = 0.31202, df = 988.22, p-value = 0.7551
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4275375  0.5892001
sample estimates:
mean of x mean of y 
 7.808642  7.727811 



	Welch Two Sample t-test

data:  data[data$cash == 0, ]$hhland_area and data[data$cash == 1, ]$hhland_area
t = -0.82084, df = 987.82, p-value = 0.4119
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -5.479340  2.247353
sample estimates:
mean of x mean of y 
  10.1444   11.7604 



	Welch Two Sample t-test

data:  data[data$cash == 0, ]$hhforest_area and data[data$cash == 1, ]$hhforest_area
t = 0.56639, df = 550.14, p-value = 0.5714
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.8489893  1.5369651
sample estimates:
mean of x mean of y 
 2.139552  1.795564 


**b) Do you conclude that the sample is well-balanced? Are you surprised or concerned by any of the results? If so, what could you do to address your concerns?**

I conclude the sample for all of them is well-balanced, because they have small p-values and T-values.

1) hhead_age --> Well balanced, has a small p-value of 0.4936 and t-value of |-0.68483|, telling us its well balanced. 

2) hhead_educ --> Well balanced, has a small p-value of 0.7551 and t-value of |0.31202|, telling us its well balanced.

3) hhland_area --> Well balanced, has a small p-value of  0.4119 and t-value of |-0.82084|, telling us its well balanced.

4) hhforest_area --> Well balanced, has a small p-value of  0.5714 and t-value of |0.56639|, telling us its well balanced.

Well balanced,Professor said, if their meann are close to one another, they are well balanced, all of them have mean that is very close to one another in the mean of x and mean of y.

### Part D

**a) What happens if you control for additional covariates (household demographics, a dummy for agreeing that environmental conservation is antithetic to their livelihood, and a dummy for having cut any trees in the past 3 years) in this regression? What (if any) advantages are there to controlling for these variables? What (if any) disadvantages are there?**

a) If you add addtional covariates, it lowers the standard errors.

B) Advtanages of adding covariaties is lowering the standard errors. If we include more covariates in our regression, we can     reduce ˆu2i , i.e. the unexplained variation in Y goes down.

C) There are no disadvantages.

**b) Run the regression controlling for all six of these variables. Interpret how your results change, if at all.** 

In [9]:
# code here
reg42<-lm(formula = forest_cover ~ cash + hhead_age + hhead_educ + hhland_area 
          + hhforest_area + any_tree_cut_3y + opi_hurtenv_agr,data = data)

In [10]:
summary(reg42)


Call:
lm(formula = forest_cover ~ cash + hhead_age + hhead_educ + hhland_area + 
    hhforest_area + any_tree_cut_3y + opi_hurtenv_agr, data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.9666  -0.0894   0.1235   0.3058  18.1769 

Coefficients:
                 Estimate Std. Error t value Pr(>|t|)  
(Intercept)     -0.198729   0.205909  -0.965   0.3347  
cash             0.192482   0.082234   2.341   0.0194 *
hhead_age        0.001569   0.002964   0.529   0.5968  
hhead_educ      -0.017750   0.010241  -1.733   0.0834 .
hhland_area     -0.002620   0.001827  -1.434   0.1519  
hhforest_area    0.007787   0.005982   1.302   0.1933  
any_tree_cut_3y -0.039036   0.119721  -0.326   0.7444  
opi_hurtenv_agr -0.292419   0.186250  -1.570   0.1167  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.29 on 982 degrees of freedom
  (5 observations deleted due to missingness)
Multiple R-squared:  0.0139,	Adjusted R-squared:  0.00686

Results should change as there are more variables/covarties.

The difference is that in the orignal we had beta0 of -0.319 and beta1(cash) at .18 and in our added covarties we have beta0 of -0.199 and beta1(cash) to be 0.19. We would reject the null hypthoesis because cash is statistical significant.This is what we expected (beta1 values) because we have a randomized control trial and so the treatment shouldn’t be significantly correlated with any of the observable characteristics. 

### Part E

**You think households might differ in the financial incentives they face when cutting trees, which might affect the effect of the cash incentives. You hypothesize that households that have been earning more money from cut trees might be less willing to stop cutting trees, so might experience smaller increases in tree cover even if they receive a cash incentive. Propose, implement, and interpret a test of whether the effect of the cash on tree cover is different for households that received above average income from cutting trees in the past year, building on the model from Part D.**

_Hint: Create the variable **above\_average\_inc** first, by summarizing the income variable, and creating a dummy variable as in Assignment 3, where the new dummy is equal to 1 when income is above the mean income. Don't forget to remove NA values from the mean calculation, using `mean(x, na.rm = TRUE)` to get the mean of x._

In [11]:
# code here
mean_income <- mean(data$log_tim_price,na.rm = TRUE)
mean_income
data$above_average_inc <-0
data$above_average_inc[data$log_tim_price > mean_income] <-1
reg123 <- lm(formula = forest_cover ~ cash + hhead_age + hhead_educ + hhland_area 
          + hhforest_area + any_tree_cut_3y + opi_hurtenv_agr + above_average_inc 
          + cash:above_average_inc, data = data)
summary(reg123)

[1] 2.401393


Call:
lm(formula = forest_cover ~ cash + hhead_age + hhead_educ + hhland_area + 
    hhforest_area + any_tree_cut_3y + opi_hurtenv_agr + above_average_inc + 
    cash:above_average_inc, data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.8336  -0.1070   0.0898   0.3591  18.1251 

Coefficients:
                        Estimate Std. Error t value Pr(>|t|)    
(Intercept)            -0.019282   0.211033  -0.091 0.927218    
cash                   -0.088705   0.115923  -0.765 0.444334    
hhead_age               0.001015   0.002954   0.344 0.731255    
hhead_educ             -0.017874   0.010186  -1.755 0.079599 .  
hhland_area            -0.002708   0.001817  -1.491 0.136394    
hhforest_area           0.008848   0.005960   1.484 0.138001    
any_tree_cut_3y         0.002034   0.129919   0.016 0.987510    
opi_hurtenv_agr        -0.290762   0.185248  -1.570 0.116834    
above_average_inc      -0.372223   0.121874  -3.054 0.002318 ** 
cash:above_average_inc  0.55974

Both above_average_inc and cash:above_average_inc show us that they are statisticaly important when it comes to determining if giving incentive to people above average income works versus those that are below average income. This regression changed it where cash isn't statistically significant anymore, telling us having above average income plays an important role when it comes to chopping down trees. 

### Part F

Of course, not everyone who was offered cash ended up taking it up and stopped cutting trees. So what we've estimated so far for the average treatment effect of _receiving cash incentives on tree cover_ can be interpreted as an **intent to treat** estimate for the effect of  _cash for environmental conservation_.

**Suppose 75\% of people who were offered cash incentives ended up stopping cutting trees, compared to 10\% of people in the control group. Write down a Treatment on the Treated estimator and calculate it using your result from Part D (you don't have to run any additional regressions for this part). Under what circumstances would this be equal to the average treatment effect of cash incentives? Do you think these hold in this context?**

In [27]:
# code here (optional)
.198729/(.75 - .10)

[1] 0.3057369

a) The treatment on the TReated estimatior would be: 

ITT = E[Complier] ∗(E[yT |Complier] −E[yC |Complier])

TOT = ITT/E[Complier]

ITT <- The beta0 intercept from part 2. and the E[Complier] is the difference between .75 - .10. 

b) For the ATE and TOT to be equal they would have unmeasured behavior that is causing that outcome, one of the ways for them to be equal is an omitted variable variable bias.


c) If you are in the treatment group, and you get money, you will wonder why someone is giving you the money, and you learn about global warming and it affects and danger, so you might stop cuttign down trees because you start to care about global warming rather than for the money. The person in the treatment group, might continue his actions as he wasn't really affected, if he was, it was because of an ommited variable.


